### DISCUSSION TITLE
Decision to Minimize Integration Support for "Commercial Products" in the Open Source Project

### DISCUSSION THREAD

**Discussion by getroot** [OvenMediaLabs/OvenMediaEngine#1634] [Announcement]
Our team and contributors have been dedicated to making OvenMediaEngine one of the best open source projects. It's truly heartening to see OvenMediaEngine being increasingly adopted. We provide extensive support within the GitHub community and offer technical assistance through private channels via email or Slack for our financial backers.

Recently, I had to make some difficult decisions for various reasons. We have decided to minimize integration support for other commercial services, software, and devices within our open source project. Generally, we aim to avoid such integrations, but we will continue to support those that are essential or easy to manage. Officially supporting commercial products requires more resources and costs than anticipated. Accessing commercial products incurs expenses, and providing reliable support demands significant manpower. As a free software project, it has become clear that properly supporting commercial products, which are paid services, is extremely challenging. Managing issues arising from integrations with commercial products is even more difficult, especially since we do not receive official support as we are not paying customers. Therefore, we have decided that our open source project will officially support only those commercial products that either collaborate with us or require minimal resources for support, or are deemed absolutely necessary.

However, we are not removing any technology from the open source. Instead, we will focus more on providing robust and stable code. For instance, in the case of integrating with DRM providers, our open source users will now need to integrate the DRM provider's API and OvenMediaEngine on their own. We will concentrate on implementing encryption algorithms and standard technologies.

OvenMediaEngine Enterprise is a product developed by a team at AirenSoft, ensuring our financial sustainability by developing commercial software based on OvenMediaEngine. They will continue to maintain such integrations with commercial services. To date, they have seamlessly integrated CloudFront, automated certificate renewals with Let's Encrypt API, and incorporated Pallycon DRM into both the Enterprise product and Open Source. Moving forward, they will focus more on Enterprise rather than integrating commercial services within the open source project. They also plan to integrate commercial cloud proxy protocols in the future.

I love the OvenMediaEngine Open Source project. My dream is to make it the world's best live streaming server and sustain it forever. Your continued contributions make this possible. This mission is my highest priority, and I am committed to making the necessary decisions to achieve it. Thank you for your ongoing support and dedication.

**Comment by bchah** [comment]
I understand and support this decision! Firstly, because some of the more vendor-specific features felt like they were for specific customers and not part of the main focus of OME. But also because I understand the struggle. My office is a gigantic mess of video switchers, encoder boxes and anything else out there that can encode/decode RTMP or SRT. The only way to be sure about how these products function with OME was to buy/subscribe to them for ongoing internal testing which is a large investment. One that is hard to be compensated for directly. So it's very important that your efforts around these commercial integrations find their rewards in the form of commercial (paid) customers. And providing the support / turn-key setup resources in your open source project is a direct conflict to that goal. 

So I am very excited for a more focused open source OME, and also for your increased commercial success with these integrations. Thank you as always to the whole Airensoft team for your hard work and dedication.

**Reply by getroot** [reply]
To avoid any misunderstandings, I want to clarify that OvenMediaEngine officially supports both RTMP and SRT. If any device or software follows the SRT/RTMP protocols but doesn’t work with OvenMediaEngine, we will do our best to support it, even if it is a commercial product.

For example, with DRM, OvenMediaEngine’s open-source version aims to fully support CENC encryption. However, we do not support the APIs of commercial DRM providers.

Similarly, with RTMP/SRT, we do support these protocols. However, we do not support the APIs of any commercial service to enable automatic streaming.

Thank you for your support
